GSE30229.Rmd

```diff
@@ -1,119 +1,217 @@
+#For adjustion we chose "BH" method because it controls the proportion of false discovery which is expected among the rejected hypotheses.
+```
+
+Did you find any probes that show statistically significant modification difference between healthy and sick individuals?
+```{r, include=TRUE, echo=TRUE}
+probes.significant <- probes[pvalues.adjusted < 0.05]
+probes.significant[1:20]
+#We choose that the adjusted p-value is "significant" if its value is below 0.05.
+```
+
+Where are these probes? What genes are they related to?
+```{r, include=TRUE, echo=TRUE}
+annotate$Gene_ID[match(probes.significant, annotate$ID)][1:20]
+annotate$CPG_ISLAND_LOCATIONS[match(probes.significant, annotate$ID)][1:20]
+#We find what genes the probes are related to and what chromosomes they are located at.
+```
+
+## Next steps
+
+Normalizavimas
+```{r, include=TRUE, echo=TRUE}
+#Normalizuojame beta verciu matrica deka normalizeBetweenArrays metodo.
+library("limma")
+normalizedMatrix <- normalizeBetweenArrays(betaValueMatrix)
+```
+
+Gauti rezultatai is https://dnamage.genetics.ucla.edu
+```{r, include=TRUE, echo=TRUE}
+#Pasileidziame gauta csv faila ir extractiname reikalingus stulpelius, kuriuos pridedame prie sickness
+researchColumns <- read.csv('GSE30229.csv')
+age_and_gender <- researchColumns[,c("DNAmAge","predictedGender")]
+age_gender_matrix <- data.matrix(age_and_gender)
+age_gender_matrix[which(age_gender_matrix[,2] == 2),2] <- "male"
+age_gender_matrix
+age_gender_matrix[which(age_gender_matrix[,2] == 1),2] <- "female"
+age_gender_matrix
+full_sickness <- cbind(sickness, age_gender_matrix)
+```
+
+Lytiniu chromosomu probu atmetimas
+```{r, include=TRUE, echo=TRUE}
+#Gauname X ir Y chromosomu indeksus anotacijos faile tam kad juos pasalinti. 
+#Taip pat pasaliname lytines chromosomas ir normalizuotoje matricoje.
+X_and_Y_indexes <- which(annotate$Chr == 'X' | annotate$Chr == 'Y')
+annotate_without_genderCh <- annotate[-(X_and_Y_indexes),]
+normalizedMatrix_without_genderCh <- normalizedMatrix[-(X_and_Y_indexes),]
+```
+
+PCA
+```{r, include=TRUE, echo=TRUE}
+source("https://bioconductor.org/biocLite.R")
+biocLite("impute")
+library("impute")
+
+#Palyginame kaip atrodys grafikas su nepasalintomis lytinemis chromosomomis ir 
+#kaip atrodys grafikas su jau pasalintomis lytinemis chromosomomis. 
+x <- 1:10
+
+imputed_1 <- impute.knn(normalizedMatrix)
+pca_1 <- prcomp(t(imputed_1$data), scale=FALSE)
+pairs(pca_1$x[,1:4], col=as.factor(full_sickness$predictedGender))
+
+imputed_2 <- impute.knn(normalizedMatrix_without_genderCh)
+pca_2 <- prcomp(t(imputed_2$data), scale=FALSE)
+pairs(pca_2$x[,1:4], col=as.factor(full_sickness$predictedGender))
+```
+Outliers
+```{r, include=TRUE, echo=TRUE}
+#Dar karta pavaizduojame grafika be lytiniu chromosomu
+normalizedMatrix_without_genderCh <- impute.knn(normalizedMatrix_without_genderCh)$data
+pca_first <- prcomp(t(normalizedMatrix_without_genderCh))
+pairs(pca_2$x[,1:4], col=as.factor(full_sickness$predictedGender))

```

Comment:
ne tas pca objektas (is tikruju rezultatas tas pats, nes tokiu paciu budu susikuret ir `pca_2` :) )

Reply:
done